plugins/by-name/fidget/deprecations.nix

```diff
@@ -0,0 +1,187 @@
+{ lib, ... }:
+let
+  inherit (lib) mkRenamedOptionModule mkRemovedOptionModule;
+  inherit (lib.nixvim) toSnakeCase;
+
+  oldPluginBasePath = [
+    "plugins"
+    "fidget"
+  ];
+  newPluginBasePath = oldPluginBasePath ++ [ "settings" ];
+
+  oldSettingsProgress = [
+    "pollRate"
+    "supressOnInsert"
+    "ignoreDoneAlready"
+    "ignoreEmptyMessage"
+    "notificationGroup"
+    "clearOnDetach"
+  ];
+  oldSettingsDisplay = [
+    "renderLimit"
+    "doneTtl"
+    "doneIcon"
+    "doneStyle"
+    "progressTtl"
+    "progressIcon"
+    "progressStyle"
+    "groupStyle"
+    "iconStyle"
+    "skipHistory"
+    "formatMessage"
+    "formatAnnote"
+    "formatGroupName"
+  ];
+  oldSettingsLsp = [
+    "progressRingbufSize"
+  ];
+  oldSettingsNotification = [
+    "pollRate"
+    "historySize"
+    "overrideVimNotify"
+  ];
+  oldSettingsView = [
+    "stackUpwards"
+    "iconSeparator"
+    "groupSeparator"
+    "groupSeparatorHl"
+  ];
+  oldSettingsWindow = [
+    "normalHl"
+    "borderHl"
+    "maxWidth"
+    "maxHeight"
+    "xPadding"
+    "yPadding"
+  ];
+  oldSettingsLogger = [
+    "floatPrecision"
+  ];
+
+  progressRenames = map (name: {
+    old = [
+      "progress"
+      name
+    ];
+    new = [
+      "progress"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsProgress;
+
+  displayRenames = map (name: {
+    old = [
+      "progress"
+      "display"
+      name
+    ];
+    new = [
+      "progress"
+      "display"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsDisplay;
+
+  lspRenames = map (name: {
+    old = [
+      "progress"
+      "lsp"
+      name
+    ];
+    new = [
+      "progress"
+      "lsp"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsLsp;
+
+  notificationRenames = map (name: {
+    old = [
+      "notification"
+      name
+    ];
+    new = [
+      "notification"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsNotification;
+
+  viewRenames = map (name: {
+    old = [
+      "notification"
+      "view"
+      name
+    ];
+    new = [
+      "notification"
+      "view"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsView;
+
+  windowRenames = map (name: {
+    old = [
+      "notification"
+      "window"
+      name
+    ];
+    new = [
+      "notification"
+      "window"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsWindow;
+
+  loggerRenames = map (name: {
+    old = [
+      "logger"
+      name
+    ];
+    new = [
+      "logger"
+      (toSnakeCase name)
+    ];
+  }) oldSettingsLogger;
```

Comment:
All of these just look like a manual way of handling what `optionsRenamedToSettings` would do for you.